### DISCUSSION TITLE
error when send number to port  /study-arm-hockey/target

### DISCUSSION THREAD

**Discussion by jimingda** [pattacini/study-air-hockey#5] [Q&A] [answered]
When i launched study-2 app and sent a number within [-1, 1] to the YARP port /study-air-hockey/target , the robot left hand move to the position as expected, but the terminal got error as below. Did i misuse the port command?
```
root@db61dac93e4e:~# yarp rpc /study-arm-hockey/target
>>0.5
[ERROR] |yarp.os.impl.BottleImpl| Reader failed, unrecognized object code 89
```

**Comment by pattacini** [comment, marked as answer]
`/study-arm-hockey/target` is a streaming port. See

https://github.com/pattacini/study-air-hockey/blob/2976edc37e798a5f28478b7e0f4eb804fad2fcea/src/study-2.cpp#L173-L176

Therefore, you ought to interact with it relying on `yarp write` as follows:
```console
$ yarp write ... /study-arm-hockey/target
0.5
```
